README.md

```diff
@@ -294,37 +294,66 @@ of that property field.
+The `_x_count` meta field that is available within `@fold` scopes represents the number of elements
+in the fold, and may be filtered or output as usual. As `_x_count` represents a count of elements,
+marking it `@output` will produce an integer value. See the [\_x_count](#_x_count) section for
+more details.
+
 #### Example Use
 ```graphql
 {
     Animal {
         name @output(out_name: "animal_name")
         out_Animal_ParentOf @fold {
+            _x_count @output(out_name: "child_count")
             name @output(out_name: "child_names")
         }
     }
 }
 ```
-Each returned row has two columns: `animal_name` with the name of each `Animal` in the graph,
-and `child_names` with a list of the names of all children of the `Animal` named `animal_name`.
-If a given `Animal` has no children, its `child_names` list is empty.
+Each returned row has three columns: `animal_name` with the name of each `Animal` in the graph,
+`child_count` with the number of children for that `Animal`, and
+`child_names` with a list of the names of all children of the `Animal` named `animal_name`.
+If a given `Animal` has no children, its `child_names` list is empty and
+the `child_count` value is 0.
 
 #### Constraints and Rules
 - `@fold` can only be applied to vertex fields, except the root vertex field.
 - May not exist at the same vertex field as `@recurse`, `@optional`, or `@output_source`.
 - Any scope that is either marked with `@fold` or is nested within a `@fold` marked scope,
   may expand at most one vertex field.
-- There must be at least one `@output` field within a `@fold` scope.
+- "No no-op `@fold` scopes": within any `@fold` scope, there must either be at least one field that
+  is marked `@output`, or there must be a `@filter` applied to the `_x_count` field.
 - All `@output` fields within a `@fold` traversal must be present at the innermost scope.
   It is invalid to expand vertex fields within a `@fold` after encountering an `@output` directive.
 - `@tag`, `@recurse`, `@optional`, `@output_source` and `@fold` may not be used anywhere
   within a scope marked `@fold`.
+- The `_x_count` meta field may only appear at the innermost scope of a `@fold` marked scope.
+- Marking the `_x_count` meta field with an `@output` produces an integer value corresponding to the
+  number of results within that fold.
+- Marking for `@output` any field other than the `_x_count` meta field produces a list of results,
+  where the number of elements in that list is equal to the value of the `_x_count` meta field,
+  if it were output.
+- If multiple fields (other than `_x_count`) are marked `@output`, the resulting output lists
+  are parallel: the `i`th element of each such list is the value of the corresponding field of
+  the `i`th element of the `@fold`, for some fixed order of elements. The order of elements
+  within the output of a `@fold` is not specified and not guaranteed to be stable between multiple
```

Comment:
"The order of elements within the output of a `@fold` is not specified"

I found this particular wording potentially confusing, since we just said "for some fixed order of elements" in the previous sentence. What if we replaced "not specified" with "fixed only for that query's particular run"?

Reply:
Yeah, agreed – it's one of the things I was hoping to get some help wordsmithing. For a given `@fold` in a given result set, for a particular execution of the query, the order is fixed. The order is not known to the user ahead of time, and may change across runs of the query, and in different result sets of a particular execution of a given query.

Reply:
To be honest, what you wrote in that comment sounds pretty clear to me ^^